graph/src/data/subgraph/schema.rs

```diff
@@ -1241,35 +1242,36 @@ pub fn attribute_index_definitions(
 ) -> Vec<AttributeIndexDefinition> {
     let mut indexing_ops = vec![];
     for (entity_number, schema_type) in document.definitions.clone().into_iter().enumerate() {
-        if let Definition::TypeDefinition(definition) = schema_type {
-            if let TypeDefinition::Object(schema_object) = definition {
-                for (attribute_number, entity_field) in schema_object
-                    .fields
-                    .into_iter()
-                    .filter(|f| f.name != "id")
-                    .enumerate()
-                {
-                    // Skip derived fields since they are not stored in objects
-                    // of this type. We can not put this check into the filter
-                    // above since that changes how indexes are numbered
-                    if is_derived_field(&entity_field) {
-                        continue;
-                    }
-                    indexing_ops.push(AttributeIndexDefinition {
-                        subgraph_id: subgraph_id.clone(),
-                        entity_number,
-                        attribute_number,
-                        field_value_type: match inner_type_name(
-                            &entity_field.field_type,
-                            &document.definitions,
-                        ) {
-                            Ok(value_type) => value_type,
-                            Err(_) => continue,
-                        },
-                        attribute_name: entity_field.name,
-                        entity_name: schema_object.name.clone(),
-                    });
+        if let Definition::TypeDefinition(TypeDefinition::Object(schema_object)) = schema_type {
+            if schema_object.name.eq(SCHEMA_TYPE_NAME) {
```

Comment:
This feels like it's going to lead to surprises for anybody who's enumerating document definitions. Can we add an extension trait for `Document` (in `graphql/src/schema/ext.rs`) that defines a function `entity_definitions(&self) -> Vec<Definition>` that can be used to iterate over 'real' definitions? Then the for loop could just be
```rust
   for (entity_number, schema_type) in document.entity_definitions().into_iter().enumerate()
```
and callers don't need to worry about special meta types